networking/ingress/getting-traffic-cluster.adoc

```diff
@@ -0,0 +1,90 @@
+[id="getting-traffic-cluster"]
+= Getting traffic into a cluster
+include::modules/common-attributes.adoc[]
+:context: getting-traffic-cluster
+
+toc::[]
+
+
+//{product-title} provides methods for communicating from
+//outside the cluster with services running in the cluster.
+
+[NOTE]
+====
+The procedures in this section require prerequisites performed by the cluster
+administrator.
+====
+
+Administrators can expose a service endpoint that external traffic can reach, by
+assigning a unique external IP address to that service from a range of external
+IP addresses.
+
+Each IP address should be assigned to only one service to ensure that each
+service has a unique endpoint. Potential port clashes are handled on a
+first-come, first-served basis.
+
+The recommendation, in order or preference, is:
+
+* If you have HTTP/HTTPS, use an ingress controller.
+* If you have a TLS-encrypted protocol other than HTTPS (for example, TLS with
+ the SNI header), use an ingress controller.
+* Otherwise, use a Load Balancer, an External IP, or a NodePort.
+
+[[external-access-options-table]]
+[options="header"]
+|===
+
+|Method |Purpose
+
+|Use an ingress controller
+|Allows access to HTTP/HTTPS traffic and TLS-encrypted protocols other than HTTPS (for example, TLS with the SNI header).
+
+|Automatically Assign a external IP Using a Load Balancer Service
+|Allows traffic to non-standard ports through an IP address assigned from a pool.
+
+|Manually assign an external IP to a service
+|Allows traffic to non-standard ports through a specific IP address.
+
+|Configure a NodePort
+|Expose a service on all nodes in the cluster.
+|===
+
+include::modules/nw-using-ingress-and-routes.adoc[leveloffset=+1]
+
+include::modules/nw-creating-external-project-and-service.adoc[leveloffset=+1]
+
+include::modules/nw-exposing-service.adoc[leveloffset=+1]
+
+include::modules/nw-using-load-balancer-getting-traffic.adoc[leveloffset=+1]
+
+include::modules/nw-using-service-external-ip.adoc[leveloffset=+1]
+
+include::modules/nw-using-nodeport.adoc[leveloffset=+1]
+
+.Prerequisites
+
+Before starting the following procedures, the administrator must:
+
+* Set up the external port to the cluster networking environment so that requests
+can reach the cluster.
+
+* Make sure there is at least one user with cluster admin role. To add this role
+to a user, run the following command:
++
+----
+oc adm policy add-cluster-role-to-user cluster-admin username
+----
+
+* Have an {product-title} cluster with at least one master and at least one node
+and a system outside the cluster that has network access to the cluster. This
+procedure assumes that the external system is on the same subnet as the cluster.
+The additional networking required for external systems on a different subnet is
+out-of-scope for this topic.
+
+include::modules/nw-assign-ips-networking.adoc[leveloffset=+1]
```

Comment:
This section, from the "Prerequisites" heading above to this `include` stanza (inclusive), can be deleted.

Reply:
On second thought, these may be relevant for the `nw-config-ip-failover` module.